src/software/ai/navigator/path_manager/path_objective.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <vector>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+
+/**
+ * Path objective is used to pass information from navigator
+ * to path manager to plan paths
+ */
+class PathObjective
+{
+   public:
+    PathObjective(const Point start, const Point end, const double current_velocity,
+                  const std::vector<Obstacle> &obstacles, RobotId robot_id)
+        : start(start),
+          end(end),
+          current_velocity(current_velocity),
+          obstacles(obstacles),
+          robot_id(robot_id)
+    {
+    }
+
+    PathObjective(const PathObjective &other)
+        : start(other.start),
+          end(other.end),
+          current_velocity(other.current_velocity),
+          obstacles(other.obstacles),
+          robot_id(other.robot_id)
+    {
+    }
+
+    const RobotId robot_id;
+    const Point start;
+    const Point end;
+    const double current_velocity;
+    const std::vector<Obstacle> obstacles;
+
+    bool operator==(const PathObjective &other) const
+    {
+        return robot_id == other.robot_id;
+    }
+
+    bool operator!=(const PathObjective &other) const
+    {
+        return robot_id != other.robot_id;
+    }
+
+    bool operator>(const PathObjective &other) const
```

Comment:
I don't really think the `PathObjective` should define `>` and `<` operators for itself within the class. The reasoning is that there is no semantic / intuitive meaning for a `PathObjective` being "less than" another `PathObjective`. I think a comparator should be created when it's needed to perform sorting operations (if that's why this operator is needed). I did something similar with `EnemyThreat` so you can use that as an example of what I mean.

Reply:
ok I made it hash instead so we can use an unordered map

Reply:
ohhh was it originally needed to be in a `set`? Makes sense, but I agree I think a hash and unordered map/set is better here.